modules/decision/decision.py

```diff
@@ -0,0 +1,109 @@
+"""
+Creates decision for next action based on current state and detected pads
+"""
+
+from .. import decision_command
+from .. import object_in_world
+from .. import odometry_and_time
+
+
+class Decision:
+    """
+    Weighs distance to located pads and variance to choose the next action, either to land or move to the nearest pad
+    """
+
+    def __init__(self, tolerance: float):
+        self.__best_landing_pad = None
+        self.__weighted_pads = []
+        self.__distance_tolerance = tolerance
+
+    @staticmethod
+    def distance_to_pad(
+        pad: object_in_world.ObjectInWorld,
+        current_position: odometry_and_time.OdometryAndTime,
+    ):
+        """
+        Calculate Euclidean distance to landing pad based on current position.
+        """
+        dx = pad.position_x - current_position.odometry_data.position.north
+        dy = pad.position_y - current_position.odometry_data.position.east
+        return (dx**2 + dy**2) ** 0.5
+
+    def __weight_pads(
+        self,
+        pads: "list[object_in_world.ObjectInWorld]",
+        current_position: odometry_and_time.OdometryAndTime,
+    ):
+        """
+        Weights the pads based on normalized variance and distance.
+        """
+        if not pads:
+            return None
+
+        distances = [self.distance_to_pad(pad, current_position) for pad in pads]
+        variances = [pad.spherical_variance for pad in pads]
+
+        max_distance = max(distances)
+
+        max_variance = max(variances)
+
+        # if max distance is 0, assumes target pad is directly below, should land
+        if max_distance == 0:
+            self.__weighted_pads = [(pads[0], 0)]
+            return None
+
+        # if all variance is 0, no pads are found
+        if max_variance == 0:
+            return None
+
+        self.__weighted_pads = [
+            (pad, distance / max_distance + variance / max_variance)
```

Comment:
It's probably worth creating a helper class to store scored landing pad information, instead of relying on an implicit data structure (tuple in this case).
```py
class ScoredLandingPad:
  def __init__(landing_pad: object_in_world.ObjectInWorld, score: float):
    self.landing_pad = landing_pad
    self.score = score
```